src/common/POLFields/FieldHolding/FieldHolding.js

```diff
@@ -0,0 +1,155 @@
+      const location = locationsMap[locationId];
+
+      setSelectedLocation(location);
+    }
+  }, [locationFieldName, locationsMap, name, values]);
+
+  const getCallNumberLabel = (callNumber = '', callNumberPrefix = '', callNumberSuffix = '') => {
+    return `${callNumberPrefix} ${callNumber} ${callNumberSuffix}`.trim();
+  };
+
+  const holdingOptions = useMemo(() => (
+    holdings?.map(({ id, permanentLocationId, callNumber, callNumberPrefix, callNumberSuffix }) => {
+      const callNumberLabel = getCallNumberLabel(callNumber, callNumberPrefix, callNumberSuffix);
+
+      return ({
+        value: id,
+        label: `${locationsMap[permanentLocationId].name} ${callNumberLabel ? ` > ${callNumberLabel}` : ''}`,
+      });
+    })
+  ), [holdings, locationsMap]);
+
+  const onChangeHolding = useCallback(
+    (holdingId) => {
+      const locationId = holdings.find(({ id }) => id === holdingId).permanentLocationId;
+
+      onChange(locationId, locationFieldName, name, holdingId);
+    },
+    [holdings, locationFieldName, name, onChange],
+  );
+
+  const selectLocationFromPlugin = useCallback(
+    (location) => {
+      setSelectedLocation(location);
+
+      onChange(location, locationFieldName, name, null);
+    },
+    [onChange, locationFieldName, name],
+  );
+
+  const clearSelectedLocation = useCallback(() => {
+    setSelectedLocation();
+    onChange(null, locationFieldName);
+  }, [locationFieldName, onChange]);
+
+  const clearButton = (
+    <IconButton
+      onClick={clearSelectedLocation}
+      icon="times-circle-solid"
+      size="small"
+    />
+  );
+
+  if (isHoldingsLoading) return <Loading />;
+
+  return (
+    selectedLocation
+      ? (isDisabled
+        ? (
+          <KeyValue
+            label={labelId ? <FormattedMessage id={locationLabelId} /> : ''}
+            value={`${selectedLocation.name}(${selectedLocation.code})`}
+          />
+        )
+        : (
+          <TextField
+            label={labelId ? <FormattedMessage id={locationLabelId} /> : ''}
+            required={required}
+            disabled
+            value={`${selectedLocation.name}(${selectedLocation.code})`}
+            endControl={isDisabled ? null : clearButton}
+          />
+        )
+      )
+      : (
+        <div>
+          <FieldSelectionFinal
+            dataOptions={holdingOptions}
+            isNonInteractive={isDisabled}
+            fullWidth
+            id={`field-${name}`}
+            label={labelId ? <FormattedMessage id={labelId} /> : ''}
+            marginBottom0
+            name={name}
+            required={required}
+            validate={required ? validateRequired : undefined}
+            onChange={onChangeHolding}
+          />
+          {!isDisabled && (
+            <LocationLookup
+              label={<FormattedMessage id="ui-orders.location.createHolding" />}
+              onLocationSelected={selectLocationFromPlugin}
+            />
+          )}
+        </div>
```

Comment:
There is a `<LocationSelection>` component in the `stripes-smart-components`.
Could it be used here?

Reply:
IMO no. It doesn't support non-interactive mode.

Reply:
`<LocationSelection>` is a standard component to make unified UI look.
For example, now has this "remote" note we added there for remote storage locations.
I don't really know is it needed here. But from general point of view, it could be.

To add `isNonInteractive` functionality, original `<LocationSelection>` could be wrapped I believe.